packages/nextjs/src/config/loaders/rollup.ts

```diff
@@ -0,0 +1,110 @@
+  // `makeAbsoluteExternalsRelative` to `false`) seems to also prevent it from going from relative to absolute in the
+  // first place, with the result that the path remains untouched (which is what we want.)
+  makeAbsoluteExternalsRelative: false,
+});
+
+const rollupOutputOptions: RollupOutputOptions = {
+  format: 'esm' as const,
+
+  // Don't create a bundle - we just want the transformed entrypoint file
+  preserveModules: true,
+};
+
+/**
+ * Use Rollup to process the proxy module file (located at `tempProxyFilePath`) in order to split its `export * from
+ * '<wrapped file>'` call into individual exports (which nextjs seems to need).
+ *
+ * @param tempProxyFilePath The path to the temporary file containing the proxy module code
+ * @param resourcePath The path to the file being wrapped
+ * @returns The processed proxy module code, or undefined if an error occurs
+ */
+export async function rollupize(tempProxyFilePath: string, resourcePath: string): Promise<string | undefined> {
+  let finalBundle;
+
+  try {
+    const intermediateBundle = await rollup(getRollupInputOptions(tempProxyFilePath, resourcePath));
+    finalBundle = await intermediateBundle.generate(rollupOutputOptions);
+  } catch (err) {
+    __DEBUG_BUILD__ &&
+      logger.warn(
+        `Could not wrap ${resourcePath}. An error occurred while processing the proxy module template:\n${err}`,
+      );
+    return undefined;
+  }
+
+  // The module at index 0 is always the entrypoint, which in this case is the proxy module.
+  let { code } = finalBundle.output[0];
+
+  // Rollup does a few things to the code we *don't* want. Undo those changes before returning the code.
+
+  // If the wrapped file only has a default export, our `export * from '...'` won't have anything in it (`export *`
+  // never includes the default export, which must be handled separately). In order to preserve whatever side effects
+  // running the file which is the source of the exports might have, though, Rollup will still emit an `import '...'
+  // statement. We're importing the same file later on in the proxy template, so we don't need this.
+  const sideEffectsImportRegex = new RegExp("import '.*';", 'g');
+  code = code.replace(sideEffectsImportRegex, '');
+
+  // Nextjs uses square brackets surrounding a path segment to denote a parameter in the route, but rollup turns those
+  // square brackets into underscores. Further, Rollup adds file extensions to bare-path-type import and export sources.
+  // Because it assumes that everything will have already been processed, it always uses `.js` as the added extension.
+  // We need to restore the original name and extension so that Webpack will be able to find the wrapped file.
+  const resourceFilename = path.basename(resourcePath);
+  const mutatedResourceFilename = resourceFilename
+    // `[\\[\\]]` is the character class containing `[` and `]`
+    .replace(new RegExp('[\\[\\]]', 'g'), '_')
```

Comment:
I think this instance of regex would benefit a lot syntax-highlighting-wise from being a literal instead of a class instance. 😄

Reply:
And of course, this is the only one I _didn't_ change. Here's both versions:

![image](https://user-images.githubusercontent.com/14812505/185527522-2d6454fe-3123-45c6-9ce4-b6d1fb11e477.png)

As you can see, the literal version actually _doesn't_ have better syntax highlighting, but that's not why I didn't change it. The reason I didn't is the linter error. It's not that it's unfixable, of course - I could just take out that backslash. But if I do that, then one of the square brackets inside the character class gets a backslash and one doesn't, which means it's way too easy to mistake the one without for a meta character rather than a literal one. Only once you notice that there are two of them in a row, and put two and two together about why the literal open bracket doesn't need to be escaped in this case, do you actually understand what the regex is saying. (You can file this under my perpetual theme of _Don't make the reader have to think if you don't have to_.)